license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/AbstractLicenseMojo.java

```diff
@@ -492,13 +500,29 @@ protected final void execute(final Callback callback) throws MojoExecutionExcept
       }
 
       // make default base dir canonical
-      this.defaultBasedir = this.getCanonicalFile(this.defaultBasedir, "license.basedir");
+      workspace.basedir = getCanonicalFile(firstNonNull(workspace.basedir, legacyDefaultBasedir), "license.workspace.basedir");
 
       // collect all the license sets together
       final LicenseSet[] allLicenseSets;
 
       // if we abandon the legacy config this contiguous block can be removed
       final LicenseSet legacyLicenseSet = convertLegacyConfigToLicenseSet();
+
+      if (workspace.basedir != null) {
+        if (legacyLicenseSet != null && legacyLicenseSet.basedir != null) {
+          if (!FileUtils.isSubfolder(legacyLicenseSet.basedir, workspace.basedir)) {
+            throw new MojoExecutionException("Legacy basedir parameter should be a subfolder of the workspace basedir.");
```

Comment:
The logic is to fail if a licenseSet baseDir is outside the workspace. 

It is perfectly legal to have only one licenset in a project with its baseDir set to the same value of the workspace baseDir